### PR TITLE
avm2+tests: Allow setting prototype to undefined

### PR DESCRIPTION
**Pull request by sleepycatcoding**
This is used internally by Adobe Flex in many places.

Progresses #12078, #12054, #11789

### CODE REVIEW

File: tests/tests/swfs/avm2/prototype_set_null/Test.as [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    import flash.display.Sprite;
+
+    public class Test extends Sprite
+    {
+    }
+}
+
+var val:Function = new Function();
+
+val.prototype = null;
```
Comment (adrian17) [inline tests/tests/swfs/avm2/prototype_set_null/Test.as:12]: cool, I had no idea this was possible :)
Could you expand the test a bit more, to make sure that instantiation works and the resulting object is usable? Like
```as
function f(){}
f.prototype = null;
var x = new f();
trace(x); // [object Object]
trace(x.asdf); // undefined
```

Comment (sleepycatcoding) [reply tests/tests/swfs/avm2/prototype_set_null/Test.as:12]: Can do.

**Comment by adrian17** [comment]
BTW minor note for the future: if an avm2 feature PR includes a test for it, then I'd just mark it as `avm2:`, no need for `+ tests` :)

File: tests/tests/swfs/avm2/prototype_set_null/Test.as (hunk 1 above)
Comment (sleepycatcoding) [reply tests/tests/swfs/avm2/prototype_set_null/Test.as:12]:
```
2023-07-24T14:21:57.091519Z  WARN run_frame: ruffle_core::display_object::movie_clip: Got AVM2 error TypeError: Cannot convert <Unknown type> to primitive.
	at /trace()
	at global$init() [TU=Test] when attempting to assign symbol class Test
```
Interestingly it fails in Ruffle at the first `trace()` call, but works in Flash Player.

Comment (sleepycatcoding) [reply tests/tests/swfs/avm2/prototype_set_null/Test.as:12]:
```
function f():void {};

f.prototype = null;

var x = new f();

trace(x.prototype); // undefined

trace(x.valueOf()); // [object Object]
trace(x.toString()); // [object Object]

trace(x); // [object Object]
trace(x.asdf); // undefined
```
It seems when `prototype` is set to undefined in Flash Player `valueOf` and `toString` methods remain, but in Ruffle they disappear?

Ruffle Output:
```
2023-07-24T14:45:23.087507Z  WARN run_frame: ruffle_core::display_object::movie_clip: Got AVM2 error TypeError: Error #1006: valueOf is not a function of class <Unknown type>.
	at global$init() [TU=Test] when attempting to assign symbol class Test
```

**Review by adrian17 — APPROVED** [review]

**Pull request merged**